lib/main.dart

```diff
@@ -1,12 +1,131 @@
 import 'package:flutter/material.dart';
+import 'data.dart';
 
 void main() => runApp(MyApp());
 
 class MyApp extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
-      home: Center(child: Text('Empty Shell for Art Museum')),
+      theme: ThemeData(primaryColor: Colors.teal[50]),
+      home: HomePage(),
+    );
+  }
+}
+
+class HomePage extends StatefulWidget {
+  @override
+  _HomePageState createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  List<ObjectInfo> picData;
+
+  void initState() {
+    super.initState();
+    requestImages().then((images) {
+      setState(() => picData = images);
+    });
+  }
+
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Colors.teal[100],
+      appBar: AppBar(
+        title: Text('The Art Museum'),
+      ),
+      body: SafeArea(
+        child: Column(
+          children: [
+            Expanded(child: worksOfArt()),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget worksOfArt() {
+    final deviceSize = MediaQuery.of(context).size;
+    return ListView.builder(
+      scrollDirection: Axis.horizontal,
+      itemCount: (picData == null) ? 0 : picData.length + 1,
+      itemBuilder: (context, index) {
+        if (index == picData.length)
+          return Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              Padding(
+                padding: const EdgeInsets.only(right: 70.0, left: 50.0),
+                child: RaisedButton(
+                  child: Text('More'),
+                  shape: StadiumBorder(),
+                  elevation: 15,
+                  onPressed: () async {
+                    final images = await requestImages();
+                    setState(() => picData.addAll(images));
+                  },
+                  color: Colors.white,
+                  splashColor: Colors.brown,
+                ),
+              ),
+            ],
+          );
+        var data = picData[index];
+        var height = deviceSize.height * .7;
+        var width = deviceSize.width * .9;
+
+        return Padding(
+          padding: const EdgeInsets.all(20.0),
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              ConstrainedBox(
```

Comment:
What is this widget for?

Reply:
The `ConstrainedBox` is how I scale the image so it fits on the screen. Before I was doing some arithmetic with the width and height but Matt suggested this to me and it works even better.

Reply:
I think there is a simpler way we could accomplish the image scaling. You could just compute the aspect ratio of the full size image, and apply that ratio to the current screen size. This could eliminate the need for extra widgetry. 

Also, this seems to create occasional misalignment in the layout, it looks like you would like the images to fall centered on the main axis, is this the aim?

I noticed some where aligned centered, while other fall above/below:
![Screen Shot 2019-08-02 at 1 59 29 PM](https://user-images.githubusercontent.com/16964204/62398859-d3ea9500-b52e-11e9-9c26-6f06da5452bf.png)

Reply:
Resolved offline